File: TYPESCRIPT/tests/main.spec.ts
import { RiceCooker } from '../main';

describe('RiceCooker', () => {
    let riceCooker: RiceCooker;

    beforeEach(() => {
        riceCooker = new RiceCooker();
    });

    it('should add rice', () => {
        riceCooker.addRice();
        expect(riceCooker['_ricePresent']).toBe(true);
    });

    it('should keep warming', () => {
        riceCooker.addRice();
        riceCooker.cookRice();
        riceCooker.keepWarm();
        riceCooker.quit();
        expect(riceCooker['_heatingInProgress']).toBe(true);
    });

});

File: RUBY/main_spec.rb
# frozen_string_literal: true

# spec/rice_cooker_spec.rb
require_relative 'main'

RSpec.describe RiceCooker do
  let(:rice_cooker) { RiceCooker.new }

  describe '#add_rice' do
    it 'adds rice when not already present' do
      rice_cooker.add_rice
      expect(rice_cooker.rice_present).to be true
    end
  end
end
